public/app/plugins/datasource/cloudwatch/components/LogGroups/LogGroupsField.test.tsx

```diff
@@ -45,6 +45,20 @@ describe('LogGroupSelection', () => {
     expect(defaultProps.onChange).toHaveBeenCalledWith([{ arn: 'arn', name: 'loggroupname' }]);
   });
 
+  it('should remove any variables before calling describeCrossAccountLogGroups', async () => {
```

Comment:
Oh yes this was confusing! I've updated the scenario name and also added a comment in the component. Hopefully that makes things clearer. 

If a template variable is defined in the legacy logs query like this:
``` typescript
const logsQuery: CloudWatchLogsQuery = {
    ...,
    logGroupNames: ['logGroup1', '$logGroupVariable'],
  }

```

Then there's no need to call the `getLogGroups` api to substitute the variable value for a log group arn - this will be taken of in the log query execution flow instead. So we just need to make sure any variables are being moved to the new log group array. So for the above query, expected output would be:
``` typescript
const logsQuery: CloudWatchLogsQuery = {
    ...,
    logGroupNames: undefined,
    logGroups: [{ arn: 'arn', name: 'loggroupname' }, { arn: '$logGroupVariable', name: '$logGroupVariable' }],
  }
```